spec/stake/arithmetic.md

```diff
@@ -0,0 +1,59 @@
+# Fixed-Point Arithmetic for Rate Computation
+
+To compute base reward rate, base exchange rate, validator-specific exchange
+rates, and total validator voting power, we need to carefully perform
+arithmetic to avoid issues with precision and rounding. We use explicitly
+specified fixed-precision arithmetic for this, with a precision of 8 digits.
+This allows outputs to fit in a u64, with all products fitting in the output
+and plenty of headroom for additions.
+
+All integer values should be interpreted as unsigned 64-bit integers, with the
+exception of the validator's commission rate, which is a `u16` specified in
+terms of basis points.
```

Comment:
Just saying that they're `u64`s isn't enough to specify the interpretation, since we also need to specify that they have an implicit denominator of $10^8$.

I think it'd be useful to add a paragraph or something describing the interpretation of a `u64` integer value, with an example of what happens during multiplication: the value $x$ is represented as the integer $x 10^8$, so the product $xy$ is represented as the integer $xy 10^8$, which we compute from the representatives $x 10^8$ and $y 10^8$ as $\lfloor (x 10^8) (y 10^8) \cdot 10^{-8} \rfloor$. 

Maybe there's a better way to explain this than the above, but it'd be useful to have an explanation of why we're dividing by 10^8, it's because when we multiply two representatives, we get an extra factor of 10^8 that we have to divide out.